Jetcaster/app/src/main/java/com/example/jetcaster/ui/home/discover/Discover.kt

```diff
@@ -38,23 +38,21 @@ import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.unit.dp
 import com.example.jetcaster.R
 import com.example.jetcaster.core.data.database.model.Category
+import com.example.jetcaster.core.data.model.FilterableCategoriesModel
+import com.example.jetcaster.core.data.model.PodcastCategoryFilterResult
 import com.example.jetcaster.designsystem.theme.Keyline1
-import com.example.jetcaster.ui.home.category.PodcastCategoryViewState
 import com.example.jetcaster.ui.home.category.podcastCategory
 
-data class DiscoverViewState(
-    val categories: List<Category> = emptyList(),
-    val selectedCategory: Category? = null
-)
-
 fun LazyListScope.discoverItems(
-    discoverViewState: DiscoverViewState,
-    podcastCategoryViewState: PodcastCategoryViewState,
+    filterableCategoriesModel: FilterableCategoriesModel,
+    podcastCategoryFilterResult: PodcastCategoryFilterResult,
     navigateToPlayer: (String) -> Unit,
     onCategorySelected: (Category) -> Unit,
     onTogglePodcastFollowed: (String) -> Unit,
 ) {
-    if (discoverViewState.categories.isEmpty() || discoverViewState.selectedCategory == null) {
+    if (filterableCategoriesModel.categories.isEmpty() ||
+        filterableCategoriesModel.selectedCategory == null
+    ) {
```

Comment:
I would suggest encapsulating this logic into the `FilterableCategoriesModel` something like 
```
val isEmptyState = categories.isEmpty() || selectedCategory == null
```

and therefore here you can check just for that boolean without worrying what's the real implementation